templates/go/client.go.twig

```diff
@@ -40,74 +116,263 @@ func (clt *Client) Set{{header.key | caseUcfirst}}(value string) {
 }
 {% endfor %}
 
+func isFileUpload(headers map[string]interface{}) bool {
+	contentType, ok := headers["content-type"].(string)
+	if ok {
+		return strings.Contains(strings.ToLower(contentType), "multipart/form-data")
+	}
+	return false
+}
+
+func (clt *Client) newfileUploadRequest(uri string, params map[string]interface{}, paramName string) (*http.Request, error) {
```

Comment:
Hey, we need a chunked and resume-able upload support here (check dart/web sdk where we have it already implemented)

Reply:
That is then something I have to implement later on. 
Due to time constraints I won't be able to do that right now. 
Should I disable the API then for now?

Reply:
@phaus no we don't need to disable the API, if it's okay with you, we can proceed with getting this merged and add the chunked upload/resumable upload later by our team or you if you get time. If that's good and some other team member review it, we can proceed forward. We are really happy with the work you have put in and the patience you had with us. Just a little more and we can get this merged 🙏🏻